torchtune/dev/grpo/rewards.py

```diff
@@ -5,17 +5,148 @@
 # LICENSE file in the root directory of this source tree.
 
 import re
-from typing import List, Tuple
+from typing import Dict, List, Optional, Protocol, Tuple, Union
 from xml.etree import ElementTree as ET
 
 import torch
-
-# from tensordict import TensorDict, TensorDictBase
-# from torchrl.data import Composite, TensorSpec, Unbounded
-# from torchrl.envs import Transform
+from dataclasses import dataclass, field
 
 from torchtune.modules.transforms.tokenizers import ModelTokenizer
 
+@dataclass
+class RewardOutput:
+    reward_base_name: str
+    total_reward: torch.Tensor
+    successes: torch.Tensor
+    # optional
+    rewards: Optional[Dict[str, torch.Tensor]] = field(default_factory=dict)
```

Comment:
I find this one confusing: what's the idea behind this? I have seen it used down in the formatter, but it seems to me that this is simply a "composite" reward. Why not simply make atomic functions? Rewards are all getting summed up anyway, so there doesn't seem any extra clarity in going with this approach?